### PR TITLE
Refine test Readme

### PR DESCRIPTION
**Pull request by xianyuanliu**
### Description
This is an update to the Readme for the test, reflecting the newly implemented test workflow and the GitHub Actions Cache system.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -19,9 +19,15 @@ These guidelines will help you to write tests to address sufficiently compact pi
 
 See more details below, particularly [test data](#test-data), [common parameters](#common-parameters), and [running tests locally](#running-tests-locally).
 
+## Test pipeline
+
+The test pipeline is defined in [`pykale/.github/workflows/test.yml`](https://github.com/pykale/pykale/blob/main/.github/workflows/test.yml). It is triggered on every pull request, push to the main branch and at midnight each day.
```
Comment (haipinglu) [inline tests/README.md:24]: say UK time

Comment (haipinglu) [reply tests/README.md:24]: It is triggered on every pull request and every push to the main branch, and we also run it at midnight (UK time) each day.

File: tests/README.md [hunk 2]
```diff
@@ -19,9 +19,15 @@ These guidelines will help you to write tests to address sufficiently compact pi
...
+
+To minimise additional downloading time, test data and pip packages once downloaded are stored in the [GitHub Actions Caches](https://github.com/pykale/pykale/actions/caches) for future reuse. Test data caching uses strict matching of the cache key, meaning any change in `tests/download_test_data.py` will update the cache. For efficiency, pip package caching uses soft matching, resulting in cache updates driven by both the date tag and any modifications in `**/setup.py`.
```
Comment (haipinglu) [inline tests/README.md:26]: minimise -->minimize
once downloaded --> ,  once downloaded, (comma to separate)

**Review by haipinglu — CHANGES_REQUESTED** [review]
Minor English problems to fix before merging. Thanks.

**Comment by xianyuanliu** [comment]
> Minor English problems to fix before merging. Thanks.

Thanks for pointing out the issues. I have made the corrections.

**Review by haipinglu — APPROVED** [review]

**Pull request merged**